transport/http/middleware_close_response_body.go

```diff
@@ -11,13 +11,13 @@ import (
 // close the response body of an operation request if the request response
 // failed.
 func AddErrorCloseResponseBodyMiddleware(stack *middleware.Stack) error {
-	return stack.Deserialize.Insert(&errorCloseResponseBodyMiddleware{}, "OperationDeserializer", middleware.Before)
+	return stack.Deserialize.Insert("OperationDeserializer", middleware.Before, &errorCloseResponseBodyMiddleware{})
```

Comment:
Since these are all begin refactored with the `Insert` method be better as the following?

```
Deserialize.Insert(Before, "Id", myMiddleware)
```